### DISCUSSION TITLE
Refetch new data showing the old data

### DISCUSSION THREAD

**Discussion by vensauro** [rrousselGit/riverpod#2712] [Q&A] [answered]
Is there a simple way for making this?
Example: 
1. open the listing page of blogs
2. open the blog
3. go back to the listing page, all results of blogs are here, but is refetched on background the data, and updated without showing a loading for the user

i tried this behaviour a while ago, and the result was a complex code and state not worthy for an app.
on the react/react native side, this is the default behaviour using react-query (tan stack query)

**Comment by rrousselGit** [comment]
There is no built-in way for this at the moment. This is planned though, but will take some time.

Isn't it the opposite flow though? As in, going back to the listing page would not re-fetch the items. But reopening the same blog-post again would refetch-it would showing previously obtained data in the meantime

Afaik refetch does not happen when going back

**Reply by vensauro** [reply]
Maybe here on riverpod, the opposite flow can work (but not showing the previously data). 
But on the other package that i made reference, [tan stack query](https://github.com/TanStack/query) the default is refetching every time the function on the component(widget) apear again.

Because is changing from inactive  to stale or fresh
Inactive: not being used on screen 
Stale: being used, but can be old data and need to be refetched next time
Fresh: being used, dont need refetch next time

i think that the apollo client can explain better about this [here](https://www.apollographql.com/docs/react/data/queries#supported-fetch-policies).
On tan stack this is more complex because relies on other things and not just one config, but have a [explanation here](https://tanstack.com/query/v5/docs/react/guides/caching)

For now, i will try on my time if i can copy this behaviour on riverpod with extension methods or another `Future`/`AsyncProvider` type.
And put here if i have some success, but is not something that i really need, is just for curiosity

**Reply by rrousselGit** [reply, marked as answer]
I am aware of the behavior. It's just that the Flutter community isn't really

Flutter folks are very used to plain old ChangeNotifiers. If we were to do this by default for Riverpod. they'd lose their ChangeNotifiers' state whenever the widget re-appear

It's something I want to cover in the long run, yes. It's just that we have to consider what's intuitive for the current community.
